src/Options.php

```diff
@@ -658,6 +659,35 @@ public function setClassSerializers(array $serializers): void
         $this->options = $this->resolver->resolve($options);
     }
 
+    /**
+     * Gets a callback that will be invoked when we sample a Transaction.
+     *
+     * @psalm-return \Sentry\Tracing\TracesSamplerInterface|callable(\Sentry\Tracing\SamplingContext): float
+     */
+    public function getTracesSampler()
+    {
+        if (\is_string($this->options['traces_sampler']) && class_exists($this->options['traces_sampler'])) {
+            return new $this->options['traces_sampler']();
+        }
+
+        return $this->options['traces_sampler'];
+    }
+
+    /**
+     * Sets a callback that will be invoked when we take the sampling decision for Transactions.
+     * Return a number between 0 and 1 to define the sample rate for the provided SamplingContext.
+     *
+     * @param callable $sampler The sampler
+     *
+     * @psalm-param \Sentry\Tracing\TracesSamplerInterface|callable(\Sentry\Tracing\SamplingContext): float $sampler
+     */
+    public function setTracesSampler($sampler): void
```

Comment:
What's the reason this option can be all those things? Wouldn't a simple `callable` be enough? If someone wants to use a class he can either code it to make it self-invokable or he can wrap the call to the class into the `callable` itself

Reply:
The reason was mostly how people like to configure these things in laravel like:
`'traces_sampler' => App\Helpers\General\TracesSampler::class`

While you are right, they can just create a callable class which might even be better since it's less code.

Reply:
Please typehint the parameter as `?callable`

Reply:
Symfony too is not so keen on using callables since it has a lot of YAML config by default. I could write a callable inside the bundle to handle FQCN string though.